Surface_mesh_segmentation/include/CGAL/internal/Surface_mesh_segmentation/SDF_calculation.h

```diff
@@ -135,9 +144,24 @@ class SDF_calculation
     normal_functor(traits.construct_normal_3_object()),
     translated_point_functor(traits.construct_translated_point_3_object()),
     centroid_functor(traits.construct_centroid_3_object()),
+    tree(create_traits(mesh, vertex_point_map)),
     use_diagonal(use_diagonal) 
   {
-    tree.insert(faces(mesh).first, faces(mesh).second, mesh, vertex_point_map);
+    typedef typename boost::property_traits<VertexPointPmap>::reference Point_ref;
+    typename GeomTraits::Collinear_3  collinear = traits.collinear_3_object();
+    face_iterator it, end;
+    for(it = faces(mesh).begin(), end = faces(mesh).end(); it!=end; it++)
+    {
+        halfedge_handle h = halfedge(*it, mesh);
+        Point_ref a(vertex_point_map[target(h, mesh)]);
```

Comment:
The syntax with `operator[]` does not work with all property maps.

It should be `get(vertex_point_map, target(h, mesh))` instead.

Cc: @sloriot